teacher.rb

```diff
@@ -1,8 +1,8 @@
 require_relative 'person'
-# Teacher class inherits from Person
+
 class Teacher < Person
-  def initialize(*_args, specialization)
-    super
+  def initialize(specialization, age, name = 'unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
```

Comment:
- [ ] like the previous CR mentioned, the user is not able to create a `teacher` the reason for this is that there is no `attr_reader` set for `specialization` if you ad this inside your teacher class it will solve the error.

Reply:
Now creates `teacher`, great implementation 👍🏽